tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py

```diff
@@ -601,10 +605,10 @@ def __call__(self, step):
           self.initial_learning_rate, name="initial_learning_rate")
       dtype = initial_learning_rate.dtype
       decay_steps = math_ops.cast(self.decay_steps, dtype)
-
+      warmup_steps = math_ops.cast(self.warmup_steps, dtype)
       global_step_recomp = math_ops.cast(step, dtype)
       global_step_recomp = math_ops.minimum(global_step_recomp, decay_steps)
-      completed_fraction = global_step_recomp / decay_steps
+      completed_fraction = (global_step_recomp - warmup_steps) / decay_steps
```

Comment:
this can be negative values? shouldn't we do tf.maximum(0, (global_step_recomp - warmup_steps))?